anndata/_core/anndata.py

```diff
@@ -492,22 +492,15 @@ def _init_as_actual(
             assert not isinstance(
                 raw, Raw
             ), "got raw from other adata but also filename?"
-            from ..readwrite.h5ad import _read_raw
-
-            raw = {
-                # Read raw dict from file
-                **_read_raw(
-                    self.file, attrs={"var", "varm"}, filename=self.filename
-                ),
-                # override retrieved entries with specified ones
-                **convert_to_dict(raw),
-            }
-        if raw and isinstance(raw, cabc.Mapping):
-            raw = Raw(self, **raw)
-        elif raw:  # is a Raw from another AnnData
-            raw = raw.copy()
-            raw._adata = self
-        self._raw = raw if raw else None
+            if {"raw", "raw.X"} & set(self.file):
+                raw = dict(X=None, **raw)
+        if not raw:
+            del self.raw
```

Comment:
What case is this?

Reply:
raw is None or raw == {} → `self._raw = None`, which I’m doing through the `fdel` API.

Reply:
But how would this be triggered? What is the case where raw has already been defined on the object, and this is how we remove it?

Reply:
It’s just `self._raw = None`.